File: xuedingtoken-frontend/src/api/proxy.ts
import { api } from './client'

export interface ChatMessage {
  role: 'system' | 'user' | 'assistant' | 'tool'
  content: string
}

export interface ChatCompletionsPayload {
  model: string
  messages: Array<{ role: string; content: string }>
  temperature?: number
  max_tokens?: number
  stream?: boolean
}

export interface ChatCompletionsResponse {
  id: string
  object: string
  created: number
  model: string
  choices: Array<{
    index: number
    message: {
      role: string
      content: string
    }
    finish_reason: string
  }>
  usage?: {
    prompt_tokens: number
    completion_tokens: number
    total_tokens: number
  }
  debug?: {
    public_model?: string
    provider_name?: string
    provider_type?: string
    provider_id?: number
    provider_key_id?: number
    policy_type?: string
    fallback_used?: boolean
    fallback_triggered?: boolean
    provider_switch_count?: number
    key_switch_count?: number
    latency_ms?: number
    cost?: number
    total_tokens?: number
    upstream_model_name?: string
    failure_chain_summary?: string
  }
}

/**
 * 调用 /proxy/chat/completions，复用 client.ts 的统一错误处理
 */
export async function chatCompletions(
  payload: ChatCompletionsPayload
): Promise<ChatCompletionsResponse> {
  return api.post<ChatCompletionsResponse>('/proxy/chat/completions', payload)
}

export async function listProxyModels(): Promise<{ object: string; data: any[] }> {
  return api.get<{ object: string; data: any[] }>('/v1/models')
}

File: backend/app/schemas/proxy.py
from pydantic import BaseModel, Field
from typing import Optional


class ChatMessage(BaseModel):
    role: str
    content: str


class ChatCompletionRequest(BaseModel):
    model: str = Field(..., description="public model name, e.g. gpt-4o")
    messages: list[ChatMessage]
    temperature: float = Field(default=0.7, ge=0, le=2)
    max_tokens: int = Field(default=1024, ge=1)
    stream: bool = Field(default=False)
    user: Optional[str] = None


class DebugInfo(BaseModel):
    public_model: Optional[str] = None
    provider_name: Optional[str] = None
    provider_type: Optional[str] = None
    provider_id: Optional[int] = None
    provider_key_id: Optional[int] = None
    policy_type: Optional[str] = None
    fallback_used: Optional[bool] = None
    fallback_triggered: Optional[bool] = None
    provider_switch_count: Optional[int] = None
    key_switch_count: Optional[int] = None
    latency_ms: Optional[float] = None
    cost: Optional[float] = None
    total_tokens: Optional[int] = None
    upstream_model_name: Optional[str] = None
    failure_chain_summary: Optional[str] = None


class ChatCompletionChoice(BaseModel):
    index: int
    message: ChatMessage
    finish_reason: str = "stop"


class ChatCompletionResponse(BaseModel):
    id: str
    object: str = "chat.completion"
    created: int
    model: str
    choices: list[ChatCompletionChoice]
    usage: Optional[dict] = None
    debug: Optional[DebugInfo] = None
